pilot/pkg/xds/endpoint_builder.go

```diff
@@ -170,12 +170,6 @@ func (b EndpointBuilder) DependentConfigs() []model.ConfigKey {
 	return configs
 }
 
-var edsDependentTypes = []config.GroupVersionKind{gvk.PeerAuthentication}
-
-func (b EndpointBuilder) DependentTypes() []config.GroupVersionKind {
-	return edsDependentTypes
```

Comment:
Isn't this breaking this? Now PA will not clear the EDS cache. They are NOT part of DependentConfigs.

DependentTypes is like DependentConfigs but when we want to say "all configs of this type", which we need for EDS.

Reply:
Yes exactly, EDS DependentTypes really shouldn't be removed, but my understanding is based on previous changes, DependentTypes in XDS cache other than EDS should be removed, because they currently provide empty interfaces, is it right?

Reply:
What do you mean 'removed'? They need the function to implement the interface. They currently just `return nil`.

What problem are you trying to solve?

Reply:
What I have some doubts is why do we need a `DependentTypes` that returns `nil`, can we remove this interface as my understanding is that `DependentConfigs` can functionally cover this part? Is the DependentTypes indispensable?

Reply:
First, why? What problems is this solving?

Second - we need it because `DependentConfigs` needs a concrete list of the exact configs, but we don't know the full list of PA required by EDS.

Reply:
> First, why? What problems is this solving?

The starting point for this PR was that I saw that `DependentTypes` had no actual return value and reference, So I wanted to remove unneeded code. But as you said, `DependentConfigs` requires a concrete list of the exact configs, so we shouldn't need to make such changes for now.

> but we don't know the full list of PA required by EDS.

And for this, I can do some investigation on PA for EDS and find out what can do to make it more clearer for us.